azkaban-common/src/main/java/azkaban/trigger/TriggerManager.java

```diff
@@ -392,6 +392,7 @@ private void checkAllTriggers() throws TriggerManagerException {
         } catch (final Throwable th) {
           //skip this trigger, moving on to the next one
           TriggerManager.logger.error("Failed to process trigger with id : " + t, th);
+          th.printStackTrace();
```

Comment:
This might be redundant if the logger already adds the stacktrace, but more importantly this is printing to stderr and not the log file configured for the logger.

Reply:
Not really, this log lib does not give us the stacktrace. an example from current web server:

> 2023/06/20 10:58:25.396 +0000 ERROR [TriggerManager] [TriggerRunnerManager-Trigger-Scanner-Thread] [Azkaban] Failed to process trigger with id : Trigger Id: 832072, Description: Trigger from SimpleTimeTrigger with trigger condition of BasicTimeChecker_1.eval() and expire condition of EndTimeChecker_1.eval(), Execute flow selfidETL from project selfid_debias_lasso_runs
> java.lang.ArrayIndexOutOfBoundsException

It's hard to investigate where this Index issue coming out.

Reply:
log4j should do the right thing, it says it will include the stacktrace:
https://logging.apache.org/log4j/2.x/log4j-api/apidocs/org/apache/logging/log4j/Logger.html#error-java.lang.CharSequence-java.lang.Throwable-
But if we are not using log4j here or it's configured differently, then just pass th.getStackTrace() concatenated with the message to the logger. th.printStackTrace() won't pipe the stacktrace to the right place, it's going to STDERR...

Reply:
make sense, updated

Reply:
I actually fix it by fillInStacktrace, we're doing the rethrowing exception a lot, might lost the stacktrace somewhere. As I've tested it I think the configs are fine and successfully print the stacktrace.